Server/ReasnAPI/ReasnAPI/Services/RoleService.cs

```diff
@@ -1,16 +1,85 @@
-﻿using ReasnAPI.Models.Database;
+﻿using Microsoft.EntityFrameworkCore;
+using ReasnAPI.Models.Database;
+using ReasnAPI.Models.DTOs;
+using System.Linq.Expressions;
 
 namespace ReasnAPI.Services {
-    public class RoleService (ReasnContext context) {
+    public class RoleService(ReasnContext context) {
         private readonly ReasnContext _context = context;
```

Comment:
Verify this warning: "Parameter 'ReasnContext context' is captured into the state of the enclosing type and its value is also used to initialize a field, property, or event." to avoid potential problems resulting from it.